wai-aria/role/invalid-roles.html

```diff
@@ -11,26 +11,109 @@
+<span role="ژ" data-testname="span with role set to Persian character (ژ)" class="ex-generic">x</span>
+<span role="ژ ♮" data-testname="span with multiple non-latin character roles, Persian character (ژ) and ♮" class="ex-generic">x</span>
+<span role="漢字" data-testname="span with role set to Japanese kanji" class="ex-generic">x</span>
+
+<!-- RTL strings (Hebrew & Arabic) with <nav> -->
+<nav role="سلام" data-testname="nav with role set to Arabic text" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="سلام حبيب" data-testname="nav with multiple role assignments set to Arabic text" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="שלום" data-testname="nav with role set to Hebrew text" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="שלום חבר" data-testname="nav with multiple role assignments set to Hebrew text" data-expectedrole="navigation" class="ex">x</nav>
+
+<!-- RTL strings (Hebrew & Arabic) with <div> -->
+<div role="سلام" data-testname="div with role set to Arabic text" class="ex-generic">x</div>
+<div role="سلام حبيب" data-testname="div with multiple role assignments set to Arabic text" class="ex-generic">x</div>
+<div role="שלום" data-testname="div with role set to Hebrew text" class="ex-generic">x</div>
+<div role="שלום חבר" data-testname="div with multiple role assignments set to Hebrew text" class="ex-generic">x</div>
+
+<!-- Escaped chars, URL-encoded chars with <a> -->
+<a role="&amp" data-testname="link with role set to ampersand character" data-expectedrole="navigation" class="ex">x</a>
+<a role="&lt" data-testname="link with role set to less than angle bracket character" data-expectedrole="navigation" class="ex">x</a>
+
+<!-- Escaped chars, URL-encoded chars with <span> -->
+<span role="&amp" data-testname="span with role set to ampersand character" class="ex-generic">x</span>
+<span role="&lt" data-testname="span with role set to less than angle bracket character" class="ex-generic">x</span>
+
+<!-- Backslash closing quote and other malformed roles with characters with <nav> -->
+<nav role="region\" data-testname="nav with role set to region followed by backslash" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="\region" data-testname="nav with role set to backslash followed by region" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="re/gion" data-testname="nav with role set to region with backslash after e character" data-expectedrole="navigation" class="ex">x</nav>
+
+<!-- Backslash closing quote and other malformed roles with characters with <span> -->
+<span role="region\" data-testname="span with role set to region followed by backslash" class="ex-generic">x</span>
+<span role="\region" data-testname="span with role set to backslash followed by region" class="ex-generic">x</span>
+<span role="re/gion" data-testname="span with role set to region with backslash after e character" class="ex-generic">x</span>
```

Comment:
typo? the last one of each of these last two blocks is a forward slash

Reply:
@cookiecrook This was intentional; do you think this is unneeded? I was curious if the role fallback worked properly when an illegal character is placed somewhere in the middle of a valid role.

Reply:
I'm suggesting that if you intend use a "/" then the test name should include "slash" or "forward slash" but not "backslash" as it's currently written.

Reply:
```suggestion
<span role="re/gion" data-testname="span with role set to region with forward slash after e character" class="ex-generic">x</span>
```